### PR TITLE
Add notifications feature

### PR DESCRIPTION
**Pull request by Schenng**
Added the Firebase Cloud Messaging Notifications.

- Added a service in order to listen for new messages while the app is backgrounded
- Added a service in order to refresh the token if expired
- Using the default notification channel, see [more](https://developer.android.com/guide/topics/ui/notifiers/notifications#ManageChannels)

Resources: 
- https://firebase.google.com/docs/cloud-messaging/android/client#manifest
- https://www.codementor.io/flame3/send-push-notifications-to-android-with-firebase-du10860kb

🎩 
1. Send a push notification from the Firebase console
screen shot 2018-11-22 at 12 10 03 am
2. Receive the message on the Android phone

### CODE REVIEW

File: GryphsWRugby/app/src/main/java/ayoolamakinde/eloisejulien/gryphswrugby/services/FirebaseIDService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+         * Modify this method to associate the user's FCM InstanceID token with any server-side account
+         * maintained by your application.
+         *
+         * @param token The new token.
+         */
+        private void sendRegistrationToServer(String token) {
+            // Add custom implementation, as needed.
+        }
+
+}
```
Comment (Schenng) [inline GryphsWRugby/app/src/main/java/ayoolamakinde/eloisejulien/gryphswrugby/services/FirebaseIDService.java:33]: This is all boilerplate, you shouldn't have to touch this

File: GryphsWRugby/app/src/main/java/ayoolamakinde/eloisejulien/gryphswrugby/services/MyFirebaseMessagingService.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @Override
+    public void onMessageReceived(RemoteMessage remoteMessage) {
+        // TODO: Handle FCM messages here.
+        // If the application is in the foreground handle both data and notification messages here.
+        // Also if you intend on generating your own notifications as a result of a received FCM
+        // message, here is where that should be initiated.
+        Log.d(TAG, "From: " + remoteMessage.getFrom());
+        Log.d(TAG, "Notification Message Body: " + remoteMessage.getNotification().getBody());
+
+    }
```
Comment (Schenng) [inline GryphsWRugby/app/src/main/java/ayoolamakinde/eloisejulien/gryphswrugby/services/MyFirebaseMessagingService.java:28]: This is where you can create custom notifications if you want

**Comment by Schenng** [comment]
@EloiseJulien I got your notification

**Comment by EloiseJulien** [comment]
Yes! I can send notifications from Firebase! Thanks :) 
I tried to send a notification when the user clicks on "send a message" but I did not succeed... 
Do you understand HTTP POST request (here https://firebase.google.com/docs/cloud-messaging/android/send-multiple?authuser=0)

In any case thank you for your help

**Pull request merged**